docs/photon/deployment.md

```diff
@@ -2,10 +2,40 @@
+connection still available.
+
+Any piece of code [outside the handler](https://docs.aws.amazon.com/lambda/latest/dg/programming-model-v2.html) remains initialized. This is a great place for
+`Photon` to call "connect" or at least call `Photon` constructor so that subsequent invocations can share a connection. There are some implications though they
+are not directly related to Photon but any system that would require a DB connection from Lambda:
+
+| Implication                                                                                                                                                                                                                                                                                                                           | Potential Solution                                                                                                                                               |
+| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| It is not guaranteed that subsequent nearby invocations of a function will hit the same container. AWS can choose to create a new container at any time.                                                                                                                                                                              | Code should assume the container to be stateless and create a connection only if it does not exist. Photon already implements that logic.                        |
+| The containers that are marked to be removed and are not being reused still keep a connection open and can stay in that state for some time (unknown and not documented from AWS), this can lead to a sub-optimal utilization of the DB connections                                                                                   | One potential solution is to use a lower idle connection timeout. Another solution can be to clean up the idle connections in a separate service<sup>1, 2</sup>. |
+| Concurrent requests might spin up separate containers i.e. new connections. This makes connection pooling a bit difficult to manage because if there is a pool of size N and C concurrent containers, the effective number of connections is N \* C. It is very easy to exhaust `max_connection` limits of the underlying data source | Photon does not implement connection pooling right now. This can also be handled by limiting the concurrency levels of a Lambda function.                        |
+
+<pre>
```

Comment:
Not a huge fan adding footnotes in a fenced code block (at least that's what the `<pre>` element is being rendered to in Markdown). Maybe `<sup>` could be an alternative? https://stackoverflow.com/questions/25579868/how-to-add-footnotes-to-github-flavoured-markdown

Reply:
Used `sup`, needed a couple of `<br />` tags to make it look good but all good.